dozer/cogs/tba.py

```diff
@@ -37,100 +41,270 @@ def __init__(self, bot):
+                await ctx.send(f"No media for team {team_num} found in {year}!")
+
+        except aiotba.http.AioTBAError:
+            raise BadArgument("Couldn't find data for team {}".format(team_num))
+
+    media.example_usage = """
+    `{prefix}tba media 971 2016` - show available media from team 971 Spartan Robotics in 2016
+    """
+
+    @tba.command()
+    @bot_has_permissions(embed_links=True)
+    async def awards(self, ctx, team_num: int, year: int = None):
+        """Gets a list of awards the specified team has won during a year. """
+        async with ctx.typing():
+            try:
+                awards_data = await self.session.team_awards(team_num, year=year)
+                events_data = await self.session.team_events(team_num, year=year)
+                event_key_map = {event.key: event for event in events_data}
+            except aiotba.http.AioTBAError:
+                raise BadArgument("Couldn't find data for team {}".format(team_num))
+
+            pages = []
+        for award_year, awards in itertools.groupby(awards_data, lambda a: a.year):
+            e = discord.Embed(title=f"Awards for FRC Team {team_num} in {award_year}:", color=discord.Color.blurple())
+            for event_key, event_awards in itertools.groupby(list(awards), lambda a: a.event_key):
+                event = event_key_map[event_key]
+                e.add_field(name=f"{event.name} [{event_key}]",
+                            value="\n".join(map(lambda a: a.name, event_awards)), inline=False)
+
+            pages.append(e)
+        if len(pages) > 1:
+            await paginate(ctx, pages, start=-1)
+        elif len(pages) == 1:
+            await ctx.send(embed=pages[0])
+        else:
+            await ctx.send(f"This team hasn't won any awards in {year}"
+                           if year is not None else "This team hasn't won any awards...yet.")
+
+    awards.example_usage = """
+    `{prefix}tba awards 1114` - list all the awards team 1114 Simbotics has ever gotten.
+    """
+
     @tba.command()
     async def raw(self, ctx, team_num: int):
         """
         Get raw TBA API output for a team.
         This command is really only useful for development.
         """
-        team_data = self.parser.get_team(team_num)
         try:
-            getattr(team_data, "Errors")
-        except tbapi.InvalidKeyError:
-            e = discord.Embed(color=blurple)
+            team_data = await self.session.team(team_num)
+            e = discord.Embed(color=discord.Color.blurple())
             e.set_author(name='FIRST® Robotics Competition Team {}'.format(team_num),
                          url='https://www.thebluealliance.com/team/{}'.format(team_num),
                          icon_url='https://frcavatars.herokuapp.com/get_image?team={}'.format(team_num))
-            e.add_field(name='Raw Data', value=team_data.flatten())
+            e.add_field(name='Raw Data', value="``` \n {}```".format(pformat(team_data.__dict__)))
```

Comment:
pformat produces Python object literals (a la `repr(...)`), so adding Python syntax highlighting will make this a bit more readable.